#!/usr/bin/env node

var options = require('../lib/options'),
    Gardener = require('../lib/main');



function usage() {
    console.log('\nUsage: gardener url [OPTIONS]\n');
    console.log('Available OPTIONS:');
    console.log(options.getHelp());
}
var args = process.argv.slice(2);

if (!args.length) {
    return usage();
}

try {
    require('npm');
} catch(e) {
    console.log('ERROR: Cant find npm');
    console.log('  solution 1: ln -s  /usr/local/lib/node_modules/npm/ node_modules/npm');
    console.log('  solution 2: npm install npm');
    return;
}


var url = args[0];
var opts = args.slice(1);
options.set_options(opts);

new Gardener(url, options).start(function(err){
    if (err) console.log(err);
});


// this is poor show.
// When couch goes down, all the follower throw an exception. Must catch it better, and in the right place

process.on('uncaughtException', function(err) {
  console.log('Caught exception: ' + err);
});
